Modules/CustomerHistory/Frames.lua

```diff
@@ -96,8 +131,13 @@ function CraftSim.CUSTOMER_HISTORY.FRAMES:SetCustomer(customer)
     end
 end
 
-function CraftSim.CUSTOMER_HISTORY.FRAMES:ResetDropdown(customer)
-    local data = CraftSim.GUTIL:Map(CraftSim.CUSTOMER_HISTORY.db.realm, function(a, e) if (a.history) then return {label=e, value=e} end end)
+function CraftSim.CUSTOMER_HISTORY.FRAMES:ResetDropdown(customer, filter)
+    local data = CraftSim.GUTIL:Map(CraftSim.CUSTOMER_HISTORY.db.realm, function(a, e)
+        valid, match = pcall(string.find, string.lower(e), string.lower(filter or ""))
```

Comment:
This again should be `local`. You should take the habit of starting to write your variables with that prefix.

Reply:
Thanks for the good catch